external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -105,34 +102,16 @@ private[kafka010] class KafkaOffsetReader(
     minPartitions.map(_ > numTopicPartitions).getOrElse(false)
   }
 
-  private def nextGroupId(): String = {
-    groupId = driverGroupIdPrefix + "-" + nextId
-    nextId += 1
-    groupId
-  }
-
   override def toString(): String = consumerStrategy.toString
 
   /**
    * Closes the connection to Kafka, and cleans up state.
    */
   def close(): Unit = {
-    if (_consumer != null) uninterruptibleThreadRunner.runUninterruptibly { stopConsumer() }
+    if (_admin != null) uninterruptibleThreadRunner.runUninterruptibly { stopAdmin() }
```

Comment:
I've already checked this deeply w/ the Kafka guys and this statement is the result of that discussion. I agree w/ the direction but I wanted to file another PR not to mix intentions. I trust the Kafka guys but I would like to execute cluster tests where I want to simulate `AdminClient` interruptions in mass.

Reply:
I've created a standalone test app which simulates mass `AdminClient` usage with interruptions: https://github.com/gaborgsomogyi/kafka-admin-interruption

I've put this into a loop and not seen any infinite lock so we can say this has proven Kafka the guys statement. I'm going to remove `UninterruptibleThread` at the end of this PR in a separate commit which is easily revertible (just in case if somebody objects it).